f8a_worker/workers/mercator.py

```diff
@@ -271,3 +275,22 @@ def get_outermost_items(list_):
             sorted_list = [i for i in sorted_list if
                            len(i['path'].split(os.path.sep)) == outermost_len]
         return sorted_list
+
+
+def validate_utf_json(result):
+    """Remove non UTF-8 characters from description if present.
+
+    :param result: dict, Dictionary having details of task
+    return: a dictionary received as input having non UTF characters removed if present
+    """
+    try:
+        if result is not None and 'details' in result and isinstance(result['details'], list):
```

Comment:
Since the UTF 8 validation is specific to mercator, it is quite evident that `details` field will always exist(See line #121) and you can skip`if result is not None and 'details' in result and isinstance(result['details'], list):` and exception handling too.
```suggestion

```